resources/lib/putio.py

```diff
@@ -205,15 +205,30 @@ def subtitles(self):
             xbmcvfs.mkdirs(subtitle_directory_path)
 
             subtitle_name = subtitle['name']
-            subtitle_language = subtitle['language_code'] or ''
-            if len(subtitle_language) == 3:  # sometimes this returns wrong
-                subtitle_name = subtitle_name + ' - ' + subtitle_language
+            if subtitle_name.endswith('.srt'):
+                subtitle_name = subtitle_name[:-4]
+
+            subtitle_lang = subtitle['language_code'] or ''
+            if len(subtitle_lang) not in (2, 3):  # sometimes this returns wrong
+                subtitle_lang = 'und'
+
+            # language is unknown for folder subtitles, use last part if name ends with `.` + 2-3 chars
+            if subtitle['source'] == 'folder':
+                parts = subtitle_name.split('.')
+                if len(parts) > 1 and len(parts[-1]) in (2, 3):
```

Comment:
This is not a check for a valid language code; any random 2-3 characters will become the "language code". Pretty sure this will yield false matches, since it comes down to the somewhat random (video and) subtitle filenaming format(s) people use.